File: Level2 - 프린터.py
def solution(p, location):
    answer = []
    loc = [x for x in range(len(p))]
    while len(p) > 0:
        if p[0] == max(p):
            p.pop(0)
            answer.append(loc.pop(0))
        else:
            p.append(p.pop(0))
            loc.append(loc.pop(0))

    return answer.index(location) + 1

File: Level2 - 프린터.js
function solution(p, l) {
  let cnt = 0;
  const loc = p.map((v) => 0);
  loc[l] = 1;

  while (p.length > 0) {
    const max = p.reduce((acc, cur) => Math.max(acc, cur));

    if (max === p[0]) {
      p.shift();
      cnt++;
      if (loc.shift() === 1) return cnt;
    } else {
      p.push(p.shift());
      loc.push(loc.shift());
    }
  }
}
